subspack.go

```diff
@@ -97,11 +97,35 @@ func (sp *SubsPack) Concatenate(sp2 *SubsPack, secPartStart time.Duration) {
 //
 // Useful if 2 different subtitles are to be displayed at the same time, e.g. 2 different languages.
 func (sp *SubsPack) Merge(sp2 *SubsPack) {
-	// Append:
-	sp.Subs = append(sp.Subs, sp2.Subs...)
-
-	// And sort
+	// Make sure inputs are properly ordered
 	sp.Sort()
+	sp2.Sort()
+
+	// Guards to prevent null pointer access
+	l1 := len(sp.Subs)
+	numSubs := l1 + len(sp2.Subs)
+
+	// Output container
+	merged := make([]*Subtitle, numSubs)
+
+	// Pointers for iteration
```

Comment:
Although I understood after I continued reading, since Golang has pointers, "Pointers" threw me off a bit at first; maybe better called here "Indices" or "Indexes" or offer futher clarification up front?